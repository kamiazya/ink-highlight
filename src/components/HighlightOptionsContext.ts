import { HighlightOptions as IHighlightOptions } from 'cli-highlight';
import { createContext } from 'react';

export type HighlightOptions = Omit<IHighlightOptions, 'theme'>;

export const HighlightOptionsContext = createContext<HighlightOptions>({});
