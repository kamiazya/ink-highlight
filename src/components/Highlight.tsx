import React, { FC, useContext } from 'react';
import PropTypes from 'prop-types';
import { highlight } from 'cli-highlight';
import { HighlightOptions, HighlightOptionsContext } from './HighlightOptionsContext';
import { HighlightThemeContext } from './HighlightThemeContext';

export type Props = {
  code: string;
} & HighlightOptions;

export const Highlight: FC<Props> = ({ code, ...option }) => {
  const theme = useContext(HighlightThemeContext);
  const contextOption = useContext(HighlightOptionsContext);
  return (
    <>
      {highlight(code, {
        ...contextOption,
        ...option,
        theme,
      })}
    </>
  );
};

Highlight.propTypes = {
  code: PropTypes.string.isRequired,
  language: PropTypes.string,
  ignoreIllegals: PropTypes.bool,
  continuation: PropTypes.any,
  languageSubset: PropTypes.arrayOf(PropTypes.string.isRequired),
};

Highlight.defaultProps = {
  language: undefined,
  ignoreIllegals: undefined,
  continuation: undefined,
  languageSubset: undefined,
};
