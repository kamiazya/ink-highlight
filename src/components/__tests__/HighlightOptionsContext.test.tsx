import React from 'react';
import { wrap } from 'jest-snapshot-serializer-raw';
import { render } from 'ink-testing-library';
import { Highlight, Props } from '../Highlight';
import { HighlightOptions, HighlightOptionsContext } from '../HighlightOptionsContext';

describe('HighlightOptionsContext', () => {
  test.each<{ attributes: Props; context: HighlightOptions }>([
    { attributes: { code: 'SELECT * FROM users;' }, context: { language: 'sql' } },
    { attributes: { code: "console.log('Hello world!!')", language: 'javascript' }, context: { language: 'sql' } },
    {
      attributes: { code: "console.log('Hello world!!')", language: 'javascript', languageSubset: ['typescript'] },
      context: {},
    },
    { attributes: { code: 'print "Hello world!"', language: 'python' }, context: { ignoreIllegals: false } },
    { attributes: { code: '<php?\n echo "Hello world!";', language: 'php' }, context: {} },
  ])('How cli-highlight is called', ({ attributes, context }) => {
    const { lastFrame } = render(
      <HighlightOptionsContext.Provider value={context}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <Highlight {...attributes} />
      </HighlightOptionsContext.Provider>,
    );
    expect(wrap(lastFrame() as string)).toMatchSnapshot();
  });
});
