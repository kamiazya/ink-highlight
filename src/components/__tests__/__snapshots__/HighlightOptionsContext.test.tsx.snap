// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightOptionsContext How cli-highlight is called 1`] = `highlight("SELECT * FROM users;", {"language":"sql"})`;

exports[`HighlightOptionsContext How cli-highlight is called 2`] = `highlight("console.log('Hello world!!')", {"language":"javascript"})`;

exports[`HighlightOptionsContext How cli-highlight is called 3`] = `highlight("console.log('Hello world!!')", {"language":"javascript","languageSubset":["typescript"]})`;

exports[`HighlightOptionsContext How cli-highlight is called 4`] = `highlight("print \\"Hello world!\\"", {"language":"python","ignoreIllegals":false})`;

exports[`HighlightOptionsContext How cli-highlight is called 5`] = `highlight("<php?\\n echo \\"Hello world!\\";", {"language":"php"})`;

exports[`HighlightOptionsContext How cli-highlight is called 6`] = `highlight("p \\"Hello world!\\"", {"continuation":{}})`;
